src/servers/src/otlp/logs.rs

```diff
@@ -0,0 +1,516 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, HashMap as StdHashMap};
+
+use api::v1::column_data_type_extension::TypeExt;
+use api::v1::value::ValueData;
+use api::v1::{
+    ColumnDataType, ColumnDataTypeExtension, ColumnOptions, ColumnSchema, JsonTypeExtension, Row,
+    RowInsertRequest, RowInsertRequests, Rows, SemanticType, Value as GreptimeValue,
+};
+use jsonb::{Number as JsonbNumber, Value as JsonbValue};
+use opentelemetry_proto::tonic::collector::logs::v1::ExportLogsServiceRequest;
+use opentelemetry_proto::tonic::common::v1::{any_value, AnyValue, InstrumentationScope, KeyValue};
+use opentelemetry_proto::tonic::logs::v1::LogRecord;
+use pipeline::{Array, Map, PipelineWay, Value as PipelineValue};
+use snafu::ResultExt;
+
+use super::trace::attributes::OtlpAnyValue;
+use crate::error::{OpenTelemetryLogSnafu, Result};
+use crate::otlp::trace::span::bytes_to_hex_string;
+
+/// Normalize otlp instrumentation, metric and attribute names
+///
+/// <https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-name-syntax>
+/// - since the name are case-insensitive, we transform them to lowercase for
+/// - better sql usability
+/// - replace `.` and `-` with `_`
+fn normalize_otlp_name(name: &str) -> String {
+    name.to_lowercase().replace(|c| c == '.' || c == '-', "_")
+}
```

Comment:
If we normalize `.` and `-` to `_`, can we transform them back to the original format when needed?

Reply:
This is the same as metric and trace. I think it can be removed with json support, but metrics and traces should be kept compatible. I'm not going to change the metrics and traces yet.

Reply:
I made a mistake, the json of the trace data doesn't do this conversion strategy. We don't need to add this conversion.